mod/primitives/pkg/hex/nibble.go

```diff
@@ -28,11 +28,11 @@ import (
 func decodeNibble(in byte) uint64 {
 	// uint64 conversion here is safe
 	switch {
-	case in >= '0' && in <= '9' && in >= hexBaseOffset:
```

Comment:
what's this change?

Reply:
hexBaseOffset is already '0' so no need for comparison here. Same with the other `if` statements

Reply:
you're right. wanna add a brief comment clarifying that this would never underflow?

Reply:
or maybe above each return, add a comment containing the expected range of the result. 

ex. `in - hexBaseOffset` would result in a value in 0-9, `in - hexAlphaOffsetUpper` would result in a value in 10-15, etc.